pkg/scheduler/queue/queue.go

```diff
@@ -277,14 +275,14 @@ func (q *RequestQueue) tryDispatchRequestToQuerier(broker *queueBroker, call *ne
 	requestSent := call.send(reqForQuerier)
 
 	if requestSent {
-		q.queueLength.WithLabelValues(string(tenantID)).Dec()
+		q.queueLength.WithLabelValues(string(tenant.tenantID)).Dec()
 	} else {
 		// should never error; any item previously in the queue already passed validation
-		err := broker.enqueueRequestFront(req)
+		err := broker.enqueueRequestFront(req, tenant.maxQueriers)
```

Comment:
this is a fix for a case introduced by a previous change that we would hit rarely, but is possible.
When maxQueriers was part of tenantRequest we could cause the following case:
1. enqueue request A, with tenant maxQueriers of 10
2. enqueue request B, including an update of tenant maxQueriers to 20, causing a resharding of the tenant queriers.
3. dequeue request A, fail to send it to the querier, and re-enqueue it, incorrectly triggering a resharding of the tenant queriers back to 10 queriers for the tenant.

So now with this fix, the maxQueriers value is decoupled from the request. The queue broker tracks the latest tenant maxQueriers value in order to provide the most up to date value for when the request needs re-enqeued after a failed dequeue.